bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
+pub struct ObjectMetadata {
+    /// Mode of this object.
+    pub mode: ObjectMode,
+
+    /// Content-Disposition of this object
+    pub content_disposition: Option<String>,
+    /// Content Length of this object
+    pub content_length: Option<u32>,
+    /// Content MD5 of this object.
+    pub content_md5: Option<String>,
+    /// Content Range of this object.
+    pub content_range: Option<Vec<u32>>,
+    /// Content Type of this object.
+    pub content_type: Option<String>,
+    /// ETag of this object.
+    pub etag: Option<String>,
+    /// Last Modified of this object.
+    pub last_modified: i64,
+}
+
+#[napi]
+pub struct ObjectLister {
+    inner: opendal::ObjectLister
+}
+
+#[napi]
+impl ObjectLister {
+    pub fn new(op: opendal::ObjectLister) -> Self {
         Self { inner: op }
     }
 
     #[napi]
-    pub async fn meta(&self, path: String) -> Result<ObjectMeta> {
-        let o = self.inner.object(&path);
-        let meta = o
+    pub async unsafe fn next_page(&mut self) -> Result<Vec<DataObject>> {
+        Ok(self.inner
+            .next_page()
+            .await
+            .map_err(format_napi_error)?.unwrap_or_default()
+            .iter()
+            .map(|obj| DataObject { inner: obj.to_owned()})
+            .collect())
+    }
+}
+
+#[napi]
+pub struct DataObject {
+    inner: opendal::Object
+}
+
+#[napi]
+impl DataObject {
+    pub fn new(op: opendal::Object) -> Self {
+        Self { inner: op }
+    }
+
+    #[napi]
+    pub async fn stat(&self) -> Result<ObjectMetadata> {
+        let meta = self.inner
             .stat()
             .await
             .map_err(format_napi_error)
             .unwrap();
 
-        let (secs, nsecs) = meta
-            .last_modified()
-            .map(|v| (v.unix_timestamp(), v.nanosecond()))
-            .unwrap_or((0, 0));
-
-        Ok(ObjectMeta {
-            location: path,
-            last_modified: DateTime::<Utc>::from_utc(
-                NaiveDateTime::from_timestamp_opt(secs, nsecs)
-                    .expect("returning timestamp must be valid"),
-                Utc,
-            ).timestamp(),
-            size: meta.content_length() as u32
-        })
+        exact_meta(meta)
     }
 
-    #[napi]
-    pub async fn write(&self,  path: String, content: Vec<u8>) -> Result<()> {
-        self.inner.object(&path)
-            .write(content)
+    #[napi(js_name="statSync")]
+    pub fn blocking_stat(&self) -> Result<ObjectMetadata> {
```

Comment:
We can use `stat_sync` to keep sync with the name in js.

Reply:
😄 I'm trying to keep the function name sync with the name in rust.

Reply:
It's a tradeoff~

Reply:
👀 Exactly, I think this is why `napi-rs` have a macro `js_name` to give an extra name.